src/test/java/com/sphy/hotelmanagementapplication/controller/HotelControllerTest.java

```diff
@@ -0,0 +1,343 @@
+                        .value("1"))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.areaName")
+                        .value("Athens"))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.disabled")
+                        .value("false"));
+
+
+        verify(hotelService, times(1)).saveHotelDTO(any());
+
+    }
+
+    @Test
+    void addHotels() throws Exception {
+        // Given
+
+
+
+        // When
+        when(hotelService.saveHotels(any())).thenReturn(hotelDTOS1);
+
+        // Return
+        mockMvc.perform(
+                        post("/api/hotels/create")
+                                .content(asJsonString(hotelDTOS1))
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .accept(MediaType.APPLICATION_JSON)
+                )
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[*]")
+                        .isArray())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[*]",hasSize(2)))
+
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[0].name")
+                        .value("hotelDTO"))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[1].name")
+                        .value("hotelDTO1"));
+
+        verify(hotelService, times(1)).saveHotels(any());
+    }
+
+    @Test
+    void findAllHotels() throws Exception {
+        // Given
+
+
+        // When
+        when(hotelService.getHotels()).thenReturn(hotelDTOS1);
+
+        // Return
+        mockMvc.perform(get("/api/hotels"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", Matchers.hasSize(2)))
+                .andExpect(jsonPath(
+                        "$[0].name",
+                        Matchers.equalTo("hotelDTO")
+                ));
+
+        // verify that roomService was executed inside findAllRooms() only once
+        verify(hotelService, times(1)).getHotels();
+    }
+
+    @Test
+    void findHotelById() throws Exception {
+        // Given
+        HotelDTO hotelDTO = new HotelDTO();
+        hotelDTO.setId(1L);
+
+        // When
+        when(hotelService.getHotelById(anyLong())).thenReturn(hotelDTO);
+
+        // Return
+        mockMvc.perform(get("/api/hotelId/1"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.id").value(1));
+
+        verify(hotelService, times(1)).getHotelById(anyLong());
+    }
+
+    @Test
+    void findHotelByName() throws Exception {
+        // Given
+        HotelDTO hotelDTO = new HotelDTO();
+        hotelDTO.setId(1L);
+        hotelDTO.setName("hotelName");
+
+        // When
+        when(hotelService.getHotelByName(anyString())).thenReturn(hotelDTO);
+
+        // Return
+        mockMvc.perform(get("/api/hotelName/hotelName"))
```

Comment:
Here, you can use the option to add a parameter, like in RoomControllerTest, line 266.